plugins/pagerduty/src/components/TriggerDialog/TriggerDialog.tsx

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useState, useEffect } from 'react';
+import {
+  Dialog,
+  DialogTitle,
+  TextField,
+  DialogActions,
+  Button,
+  DialogContent,
+  Typography,
+  CircularProgress,
+} from '@material-ui/core';
+import { useApi, alertApiRef, identityApiRef } from '@backstage/core';
+import { useAsyncFn } from 'react-use';
+import { pagerDutyApiRef } from '../../api';
+import { Alert } from '@material-ui/lab';
+
+type Props = {
+  name: string;
+  integrationKey: string;
+  showDialog: boolean;
+  handleDialog: () => void;
+  setShouldRefreshIncidents: (shouldRefreshIncidents: boolean) => void;
+};
+
+export const TriggerDialog = ({
+  name,
+  integrationKey,
+  showDialog,
+  handleDialog,
+  setShouldRefreshIncidents,
```

Comment:
This should be an `onAlertTriggered: () => void` and let the caller deal with that semantic thing happening